app.ino

```diff
@@ -203,62 +203,114 @@ void blinkLED(int times)
-  float temperature = 0;
-  pressureSensor -> getPressure(&pressure);
-  pressureSensor -> getTemperature(&temperature);
-  char buff[128];
-  sprintf(buff, "Environment\r\nPressure: \r\n    %shPa\r\nTemp: %sC \r\n",f2s(pressure, 2), f2s(temperature, 1));
-  Screen.print(buff);
-}
-
-void update(){
-    switch(keepUpdate) {
-        case 1: showMotionGyroSensor();break;
-        case 2: showPressureSensor();break;
-        case 3: showHumidTempSensor();break;
-        // case 4: showMagneticSensor();break;
-    }
+    float pressure = 0;
+    float temperature = 0;
+    pressureSensor->getPressure(&pressure);
+    pressureSensor->getTemperature(&temperature);
+    char buff[128];
+    sprintf(buff, "Environment\r\nPressure: \r\n    %shPa\r\nTemp: %sC \r\n", f2s(pressure, 2), f2s(temperature, 1));
+    Screen.print(buff);
 }
 
 void showHumidTempSensor()
 {
-  ht_sensor->reset();
-  float temperature = 0;
-  ht_sensor->getTemperature(&temperature);
-  //convert from C to F
-  temperature = temperature*1.8 + 32;
-  float humidity = 0;
-  ht_sensor->getHumidity(&humidity);
-
-  char buff[128];
-  sprintf(buff, "Environment \r\n Temp:%sF    \r\n Humidity:%s%% \r\n          \r\n",f2s(temperature, 1), f2s(humidity, 1));
-  Screen.print(buff);
+    ht_sensor->reset();
+    float temperature = 0;
+    ht_sensor->getTemperature(&temperature);
+    //convert from C to F
+    temperature = temperature * 1.8 + 32;
+    float humidity = 0;
+    ht_sensor->getHumidity(&humidity);
+
+    char buff[128];
+    sprintf(buff, "Environment \r\n Temp:%sF    \r\n Humidity:%s%% \r\n          \r\n", f2s(temperature, 1), f2s(humidity, 1));
+    Screen.print(buff);
 }
 
 void showMagneticSensor()
 {
-  magnetometer->getMAxes(axes);
-  char buff[128];
-  sprintf(buff, "Magnetometer  \r\    x:%d     \r\n    y:%d     \r\n    z:%d     ", axes[0], axes[1], axes[2]);
-  Screen.print(buff);
+    magnetometer->getMAxes(axes);
+    char buff[128];
+    sprintf(buff, "Magnetometer  \r\    x:%d     \r\n    y:%d     \r\n    z:%d     ", axes[0], axes[1], axes[2]);
+    Screen.print(buff);
+}
+
+void updateSensor()
+{
+    switch (updateSensorIndicator)
+    {
+    case 1:
+        showMotionGyroSensor();
+        break;
+    case 2:
+        showPressureSensor();
+        break;
+    case 3:
+        showHumidTempSensor();
+        break;
+        // case 4: showMagneticSensor();break;
+    }
+}
+
+void setup()
+{
+    Screen.clean();
+    Screen.print(2, "Initializing...");
+    Screen.print(3, " > WiFi");
+
+    ext_i2c = new DevI2C(D14, D15);
+    acc_gyro = new LSM6DSLSensor(*ext_i2c, D4, D5);
+    acc_gyro->init(NULL);
+    acc_gyro->enableAccelerator();
+    acc_gyro->enableGyroscope();
+
+    ht_sensor = new HTS221Sensor(*ext_i2c);
+    ht_sensor->init(NULL);
+
+    magnetometer = new LIS2MDLSensor(*ext_i2c);
+    magnetometer->init(NULL);
+
+    pressureSensor = new LPS22HBSensor(*ext_i2c);
+    pressureSensor->init(NULL);
+
+    hasWifi = false;
+    initWifi();
```

Comment:
hasWifi = (WiFi.begin() == WL_CONNECTED);
if(!hasWifi)
{}
Then we don't need the initWifi() function.